Dockerfile

```diff
@@ -1,59 +1,50 @@
-#RUN rpm -iUvh docker-engine-${DOCKER_VERSION}-1.el7.centos.x86_64.rpm
-ENV DVM_VERSION 0.6.4
-
-RUN curl -s https://raw.githubusercontent.com/getcarina/dvm/${DVM_VERSION}/install.sh | sh && \
-  source /root/.dvm/dvm.sh && \
-  dvm install ${DOCKER_VERSION}
-
+ENV DOCKER_VERSION v17.03.1-ce
+RUN curl -s https://raw.githubusercontent.com/howtowhale/dvm/master/install.sh | sh && source /root/.dvm/dvm.sh && dvm install ${DOCKER_VERSION}
 RUN echo source /root/.dvm/dvm.sh >> /root/.bashrc && echo dvm install ${DOCKER_VERSION} >> /root/.bashrc
-#RUN yum update && yum install -y docker-engine-${DOCKER_VERSION}-1.el7.centos
+
+# Install docker-compose
+RUN pip install docker-compose
 
 # Install kubectl
-ENV KUBE_VER=1.4.3
+ENV KUBE_VER=1.6.0
 ENV KUBE_URL=https://storage.googleapis.com/kubernetes-release/release/v${KUBE_VER}/bin/linux/amd64/kubectl
-RUN /bin/bash -l -c "wget --quiet ${KUBE_URL} \
-                     -O /usr/local/bin/kubectl && \
-                     chmod +x /usr/local/bin/kubectl"
+RUN /bin/bash -l -c "wget --quiet ${KUBE_URL} -O /usr/local/bin/kubectl && chmod +x /usr/local/bin/kubectl"
 
 # Install S3 Secrets
 RUN /usr/bin/mkdir -p /opt/bin
 RUN URL=https://github.com/UKHomeOffice/s3secrets/releases/download/v0.1.3/s3secrets_v0.1.3_linux_x86_64 OUTPUT_FILE=/opt/bin/s3secrets MD5SUM=ec5bc16e6686c365d2ca753d31d62fd5 /usr/bin/bash -c 'until [[ -x ${OUTPUT_FILE} ]] && [[ $(md5sum ${OUTPUT_FILE} | cut -f1 -d" ") == ${MD5SUM} ]]; do wget -q -O ${OUTPUT_FILE} ${URL} && chmod +x ${OUTPUT_FILE}; done'
 
-# Install docker-compose
-RUN URL=https://github.com/docker/compose/releases/download/1.7.1/docker-compose-Linux-x86_64 OUTPUT_FILE=/usr/local/bin/docker-compose MD5SUM=d7e662ce0f9833ac6c28a07fcb169097 /usr/bin/bash -c 'until [[ -x ${OUTPUT_FILE} ]] && [[ $(md5sum ${OUTPUT_FILE} | cut -f1 -d" ") == ${MD5SUM} ]]; do wget -q -O ${OUTPUT_FILE} ${URL} && chmod +x ${OUTPUT_FILE}; done'
-
+# Configure Jenkins and deploy latest compatible plugins with $JENKINS_VERSION
 ENV JENKINS_HOME /var/lib/jenkins
-
+ENV JENKINS_UC https://updates.jenkins.io
 ADD jenkins.sh /srv/jenkins/jenkins.sh
 ADD jenkins_backup.sh /srv/jenkins/jenkins_backup.sh
-
-# User config / updates
-# JENKINS_UC is needed to download plugins
-ENV JENKINS_UC https://updates.jenkins.io
 COPY plugins.sh /usr/local/bin/plugins.sh
 COPY plugins.base.txt /usr/share/jenkins/ref/
-RUN /usr/local/bin/plugins.sh /usr/share/jenkins/ref/plugins.base.txt
 COPY plugins.txt /usr/share/jenkins/ref/
-RUN /usr/local/bin/plugins.sh /usr/share/jenkins/ref/plugins.txt
+
+# Until keycloak plugin is available via the update center, it has to be built from the source.
+RUN /usr/local/bin/plugins.sh /usr/share/jenkins/ref/plugins.base.txt && \
+	/usr/local/bin/plugins.sh /usr/share/jenkins/ref/plugins.txt && \
+	git --version && mvn --version && \
+	git clone https://github.com/jenkinsci/keycloak-plugin.git && \
```

Comment:
We shouldn't really be compiling plugins in the dockerfile, can you not just install the plugin via file upload in the gui for the time being?